src/mono/wasm/runtime/binding_support.js

```diff
@@ -2115,12 +2375,208 @@ var BindingSupportLib = {
 
 			return 0;
 		} catch (exc) {
-			return BINDING.js_string_to_mono_string(exc.message);
+			return BINDING.wrap_error(null, exc.message);
 		} finally {
 			nameRoot.release();
 		}
 	},
+	mono_wasm_cancel_promise: function (thenable_js_handle, is_exception) {
+		try {
+			const promise = BINDING.mono_wasm_get_jsobj_from_js_handle(thenable_js_handle)
+			const promise_control = promise[BINDING.promise_control_symbol];
+			promise_control.reject("OperationCanceledException");
+		}
+		catch (ex) {
+			return BINDING.wrap_exception(is_exception, ex);
+		}
+	},
+	mono_wasm_web_socket_open: function (uri, subProtocols, on_close, web_socket_js_handle, thenable_js_handle, is_exception) {
+		var uri_root = MONO.mono_wasm_new_root(uri);
+		let sub_root = MONO.mono_wasm_new_root(subProtocols)
+		let on_close_root = MONO.mono_wasm_new_root(on_close)
+		try {
+			const js_uri = BINDING.conv_string(uri_root.value);
+			if (!js_uri) {
+				return BINDING.wrap_error(is_exception, "ERR12: Invalid uri '" + uri_root.value + "'");
+			}
+
+			const js_subs = BINDING._mono_array_root_to_js_array(sub_root);
+
+			const js_on_close = BINDING._wrap_delegate_root_as_function(on_close_root)
+
+			const ws = new globalThis.WebSocket(js_uri, js_subs);
+			var { promise, promise_control } = BINDING._create_cancelable_promise();
+
+			ws[BINDING.wasm_ws_pending_receive_event_queue] = new BINDING.Queue();
+			ws[BINDING.wasm_ws_pending_receive_promise_queue] = new BINDING.Queue();
+			ws[BINDING.wasm_ws_pending_open_promise] = promise_control;
+			ws[BINDING.wasm_ws_pending_send_promises] = [];
+			ws.binaryType = "arraybuffer";
+			const on_message = (ev) => {
+				BINDING._mono_wasm_web_socket_on_message(ws, ev)
+
+				if (BINDING.isChromium) {
```

Comment:
We may want to put the isChromium check in prevent_timer_throttling instead of sprinkling it everywhere. Otherwise if we expand the throttling management to cover Firefox as well, we have to go change all these if statements.